src/pages/checkout/index.tsx

```diff
@@ -141,6 +177,25 @@ const Checkout: React.FC = () => {
 				}
 			});
 	};
+
+	const handleDiscount = (e: React.MouseEvent<HTMLButtonElement>) => {
+		api
+			.usersShoppingCartCouponApply({ code: inputPromoValue })
+			.then((data) => {
+				setDiscount(data);
+				setPromocodeError('');
+			})
+			.catch((error) => {
+				setDiscount(discountInitial);
```

Comment:
В продолжение предложения выше:

```suggestion
				setDiscount(null);
```